### DISCUSSION TITLE
How do i sync motors in pybricks?

### DISCUSSION THREAD

**Discussion by lee-jingyuan** [pybricks/support#1949] [Q&A]
Im participateing in FLL this year and I'm trying to sync motors in python but I'm not sure how?

**Comment by BertLindeman** [comment]
Hello,

Do you mean sync as in: rotate to the same angle?
As an example like this:
```python
motor_left.run_angle(motor_speed, target_angle, wait=False)
motor_right.run_angle(motor_speed, target_angle, wait=False)
```

Is that what you meant?

Bert

**Reply by lee-jingyuan** [reply]
Kinda. I also want it to turn at the same speed in the same time even if the motors are not paired

**Reply by BertLindeman** [reply]
Than have a look at drivebase. That does the sync if you have it do drive straight.
Even if the motors are in a drive base you can control them on each own.

Bert

**Reply by lee-jingyuan** [reply]
Hi. Is it possible to control the individual speed of the motor while using drive base.

**Reply by BertLindeman** [reply]
> Hi. Is it possible to control the individual speed of the motor while using drive base.

Yes

**Reply by BertLindeman** [reply]
A simple example (I wanted to be sure):
```python

from pybricks.robotics import DriveBase
from pybricks.pupdevices import Motor
from pybricks.parameters import Port

motor_a = Motor(Port.A)
motor_b = Motor(Port.B)
my_base = DriveBase(motor_a, motor_b, 50, 180)  # just filled something is

my_base.straight(200)
motor_a.stop()
motor_b.run_angle(100, -180)

motor_a.run_angle(100, 180)
```

Bert
